speculation-rules.bs

```diff
@@ -398,7 +401,10 @@ add the following step
     1. If |clause| is null, then return null.
     1. Return a [=document rule negation=] whose [=document rule negation/clause=] is |clause|.
   1. If |predicateType| is "`href_matches`":
-    1. If |input| has any [=map/key=] other than "`href_matches`", then return null.
+    1. If |input| has any [=map/key=] other than "`href_matches`" and "`relative_to`", then return null.
+    1. If |input|["`relative_to`"] [=map/exists=], then:
```

Comment:
The case I'm not super sure about is:

```json
{
   "source": "document",
   "where": {"or": [{"href_matches": "/foo"}, {"href_matches": "/bar", "relative_to": "document"}]},
   "relative_to": "document"
}
```

It feels like the the speculation rule level `relative_to` here was probably accidentally added and could even be a bug, should we fail parsing to guard against this?

Reply:
Yeah, it's ambiguous and kinda weird since we haven't yet defined a keyword to change it back. I think my gut is to discard any document rules that use relative_to at the rule level, as invalid. We could always add it in the future if necessary without breaking compat with any valid rules.

Reply:
Yeah, that sounds reasonable to me. Updated the spec and and explainer to account for this.